contracts/mapping-token/contracts/v2/Backing.sol

```diff
@@ -0,0 +1,268 @@
+        emit NewInBoundLaneAdded(mappingTokenFactory, inboundLane);
+    }
+
+    // here add OutBoundLane, and remote issuing module must add the corresponding InBoundLane
+    function addOutboundLane(address outboundLane) external onlyOwner {
+        uint32 bridgedChainPosition = IMessageVerifier(outboundLane).bridgedChainPosition();
+        require(remoteChainPosition == bridgedChainPosition, "Invalid bridged chain position");
+        uint32 bridgedLanePosition = IMessageVerifier(outboundLane).bridgedLanePosition();
+        outboundLanes[bridgedLanePosition] = outboundLane;
+        emit NewOutBoundLaneAdded(bridgedLanePosition, outboundLane);
+    }
+
+    /**
+     * @notice reigister new erc20 token to the bridge. Only owner can do this.
+     * @param bridgedLanePosition the bridged lane positon, this register message will be delived to this lane position
+     * @param token the original token address
+     * @param name the name of the original token
+     * @param symbol the symbol of the original token
+     * @param decimals the decimals of the original token
+     */
+    function registerErc20Token(
+        uint32 bridgedLanePosition,
+        address token,
+        string memory name,
+        string memory symbol,
+        uint8 decimals
+    ) external payable onlyOwner {
+        require(registeredTokens[token] == false, "Backing: token has been registered");
+
+        address outboundLane = outboundLanes[bridgedLanePosition];
+        require(outboundLane != address(0), "cannot find outboundLane to send message");
+        bytes memory newErc20Contract = abi.encodeWithSelector(
+            IMappingTokenFactory.newErc20Contract.selector,
+            address(this),
+            ERC20_TOKEN_TYPE,
+            token,
+            thisChainName,
+            name,
+            symbol,
+            decimals
+        );
+        uint256 fee = IFeeMarket(feeMarket).market_fee();
+        require(msg.value >= fee, "not enough fee to pay");
+        uint64 nonce = IOutboundLane(outboundLane).send_message{value: fee}(remoteMappingTokenFactory, newErc20Contract);
+        uint256 messageId = encodeMessageId(bridgedLanePosition, nonce);
+        registerMessages[messageId] = token;
+        if (msg.value > fee) {
+            require(payable(msg.sender).send(msg.value - fee), "transfer back fee failed");
+        }
+        emit NewErc20TokenRegistered(messageId, bridgedLanePosition, token);
+    }
+
+    /**
+     * @notice lock original token and issuing mapping token from bridged chain
+     * @dev maybe some tokens will take some fee when transfer
+     * @param bridgedLanePosition the bridged lane positon, this issuing message will be delived to this lane position
+     * @param token the original token address
+     * @param recipient the recipient who will receive the issued mapping token
+     * @param amount amount of the locked token
+     */
+    function lockAndRemoteIssuing(
+        uint32 bridgedLanePosition,
```

Comment:
Can we keep the user unaware of the existence of the lane, and let the `Backing` contract manage the specific use of that lane?

Reply:
For example, adding a logical contract to handle the transaction procedures, to handle tokens that require fees, because even if these details are handled here, there are still many tokens whose specific logic we cannot envision, such as the rebase token, whose balance will deflate or inflate over time

Reply:
this will allow user use different LanePosition to send message